pkg/metallb/frr_test.go

```diff
@@ -0,0 +1,321 @@
+			expectedError: "frrConfiguration 'connectTime' value is not valid",
+		},
+	}
+
+	for _, testCase := range testCases {
+		frrConfigurationBuilder := testCase.testFrrConfiguration.WithBGPRouter(64550).
+			WithBGPNeighbor("10.46.73.131", 64500).
+			WithConnectTime(*testCase.connectTime, 0)
+		assert.Equal(t, testCase.expectedError, frrConfigurationBuilder.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.connectTime,
+				frrConfigurationBuilder.Definition.Spec.BGP.Routers[0].Neighbors[0].ConnectTime)
+		}
+	}
+}
+
+func TestFrrConfigurationWithEBGPMultiHop(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		ebgpMultiHop         bool
+		expectedError        string
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			ebgpMultiHop:         true,
+			expectedError:        "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		frrConfigurationBuilder := testCase.testFrrConfiguration.WithBGPRouter(64550).
+			WithBGPNeighbor("10.46.73.131", 64500).WithEBGPMultiHop(0)
+		assert.Equal(t, testCase.expectedError, frrConfigurationBuilder.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.ebgpMultiHop,
+				frrConfigurationBuilder.Definition.Spec.BGP.Routers[0].Neighbors[0].EBGPMultiHop)
+		}
+	}
+}
+
+func TestFrrConfigurationWithToReceiveModeAll(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		mode                 string
+		expectedError        string
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			mode:                 "all",
+			expectedError:        "",
+		},
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			mode:                 "allowAll",
+			expectedError:        "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		frrConfigurationBuilder := testCase.testFrrConfiguration.WithBGPRouter(64550).
+			WithToReceiveModeAll(testCase.mode, 64550)
+		assert.Equal(t, testCase.expectedError, frrConfigurationBuilder.errorMsg)
+
+		if testCase.expectedError == "all" {
+			assert.Equal(t, testCase.mode,
+				frrConfigurationBuilder.Definition.Spec.BGP.Routers[0].Neighbors[0].ToReceive.Allowed.Mode)
+		}
+	}
+}
+
+func TestFrrConfigurationWithToReceiveModeFiltered(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		prefix               string
+		expectedError        string
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			prefix:               "192.168.1.0/24",
+			expectedError:        "",
+		},
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			prefix:               "192.0/24",
+			expectedError:        "",
```

Comment:
No error message? not sure what're we testing here

Reply:
Yes it is incorrect it is meant to check this error The peerIP for the frrConfiguration bgp peer contains invalid ip address. Fixed